### DISCUSSION TITLE
How to use graphql with js-sdk

### DISCUSSION THREAD

**Discussion by DeanKamali** [directus/directus#10490] [Archive]
Playing around with the SDK, I found graphql method, when attempting to run a query that works in my Desktop client, I get `GraphQL schema validation error.`; 

I couldn't find examples in the docs. 

Any thoughts on why it's not working? 

```js
   let data = await directus.graphql.items(`
        query {
            roles(filter: { id: { _eq: "307d38df-5113-4ed5-80d0-d299c5b1e548"} }) {
            id
            name
            icon
            description
            ip_access
            enforce_tfa
            admin_access
            app_access
            users {
            id
            first_name
            last_name
            }
            } 
        }`); 
```

**Comment by joselcvarela** [comment]
Hello @DeanKamali 
To access `directus_roles` you will need to use `directus.graphql.system('query{...}')`. 
Can you try it?

**Reply by DeanKamali** [reply]
@joselcvarela That worked, thank you
